### PR TITLE
feat: add helpers to get module values on scans

### PR DESCRIPTION
**Pull request by vthib**
I need to be able to read and enumerate the module values that YARA generated during a scan. This can be done by listening for the ModuleImported callback msg, and going through the object given by YARA.

This MR adds helper to be able to do this with Rust bindings:

- Handle the "ModuleImported" scan message: this returns a pointer to a YR_OBJECT.

- Add helpers to investigate the values stored in a YR_OBJECT. This is done lazily, so instead of converting a YR_OBJECT to a rust version, the values are converted when the `object.value()` function is called.

This isn't the cleaner code i've ever written, but the yara code is quite messy as well. the YR_OBJECT is a "simulated OOP in C"-style struct, and like all OOP impls in C, it is terribly unsafe. So I have used a big unsafe block when converting the YR_OBJECT to a rust enum (YrObjectValue), which does the same thing that is done on yara's side when it is reading those values. I could remove this huge unsafe block and replace it with multiple local unsafe blocks, but i'm not sure how much it would really improve this code.

Please tell me what you think, thanks!

### CODE REVIEW

File: src/internals/object.rs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                    let len = (*this.items).used;
+
+                    YrObjectValue::Dictionary(
+                        (0..len)
+                            .filter_map(|i| {
+                                let v = (*this.items).objects[i as usize];
+
+                                if v.key.is_null() || v.obj.is_null() {
+                                    return None;
+                                }
```
Comment (Hugal31) [inline src/internals/object.rs:122]: Do we want to filter out the null values?

Comment (vthib) [reply src/internals/object.rs:122]: Those two pointers are not supposed to be NULL, in libyara, it derefs those without checks.
I added those checks to be safer, and avoid null dereferences leading to UB

File: src/internals/object.rs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                    if this.items.is_null() {
+                        return YrObjectValue::Dictionary(HashMap::new());
+                    }
+
+                    let len = (*this.items).used;
+
+                    YrObjectValue::Dictionary(
+                        (0..len)
+                            .filter_map(|i| {
+                                let v = (*this.items).objects[i as usize];
```
Comment (vthib) [inline src/internals/object.rs:118]: There is an issue here, objects is declared with length 1 in "YR_OBJECT_DICTIONARY", but its a variable-sized struct. Need to rebuild a slice from the length like is done for the array case.

**Comment by vthib** [comment]
Linter is failing but unrelated to this MR: rust 1.67 just got released and some new lints are reported on existing code.

**Comment by vthib** [comment]
Anything blocking this MR, or that you want me to change @Hugal31 ?

**Comment by Hugal31** [comment]
Nop, it's fine. Thank you!

**Pull request merged**